journal/fs.go

```diff
@@ -42,6 +46,8 @@ func OpenFSJournal(lr repo.LockedRepo, disabled DisabledEvents) (Journal, error)
 		EventTypeRegistry: NewEventTypeRegistry(disabled),
 		dir:               dir,
 		sizeLimit:         1 << 30,
+		keep:              3,
+		old:               list.New(),
```

Comment:
You're probably right. Was trying to avoid the small memory leak that would occur with append, but switched instead to keeping track of the index and rotating through the index. No good reason for sure.